client/jetpack-cloud/components/guided-tour/index.tsx

```diff
@@ -10,16 +10,39 @@ import { preferencesLastFetchedTimestamp } from 'calypso/state/preferences/selec
 import './style.scss';
 
 export interface Tour {
-	targetId: string;
+	target: string;
 	title: string;
 	description: string;
+	popoverPosition?:
+		| 'top'
+		| 'top right'
+		| 'right'
+		| 'bottom right'
+		| 'bottom'
+		| 'bottom left'
+		| 'left'
+		| 'top left';
 }
 
 interface Props {
 	tours: Tour[];
 	preferenceName: string;
 }
 
+// Helper function to get an element, returning a promise that resolves when the element is rendered.
+const getElementAsync = ( target: string ): Promise< HTMLElement > =>
+	new Promise( ( resolve ) => {
+		const getElement = () => {
+			const element = document.querySelector( target ) as HTMLElement | null;
+			if ( element ) {
+				resolve( element );
+			} else {
+				requestAnimationFrame( getElement );
+			}
+		};
+		getElement();
+	} );
```

Comment:
Thinking out loud: I wonder if this can be written as a custom hook that calls `useEffect`, makes a ref with `useRef` to track whether or not the effect has triggered, and keeps a counter with `useState` to limit the number of attempts. This would also remove the need for a `Promise`, since the effect executes only when the page changes, and only for a maximum number of tries.

Am I making this too complex?

Reply:
This seems like the better approach. Let me see how I can work on this 👍

Reply:
> Is there ever a chance that the target element never shows up? And would this cause performance problems if we continually request new animation frames?

Right now, all the targeted elements are expected to show, but I do understand your concern. I've added some timeout so the function will not endlessly query for a component that will never appear. Thanks for raising this concern.

Reply:
@mattgawarecki I couldn't figure out a way to utilize ref with `useRef` without changing how `GuidedTour` component works. Although I did convert the helper function to a react hook and avoid using `Promise` instead.